filament/backend/src/opengl/OpenGLDriver.h

```diff
@@ -373,8 +386,16 @@ class OpenGLDriver final : public backend::DriverBase {
     void setExternalTexture(GLTexture* t, void* image);
 
     void whenGpuCommandsComplete(std::function<void()> fn) noexcept;
+    void whenFrameBegins(std::function<bool()> fn) noexcept;
+
     void executeGpuCommandsCompleteOps() noexcept;
-    std::vector<std::pair<GLsync, std::function<void(void)>>> mGpuCommandCompleteOps;
+    void executeFrameBeginsOps() noexcept;
+
+    std::vector<std::pair<GLsync, std::function<void()>>> mGpuCommandCompleteOps;
+    std::vector<std::function<bool()>> mFrameBeginsOps;
+
+    TimerQueryInterface* mTimerQuery = nullptr;
```

Comment:
To avoid confusion with the _actual_ timer queries, maybe rename to something like `mTimerQueryStrategy` or `mTimerQueryImpl`?

Reply:
sounds good